### PR TITLE
This PR adds the requirements midnight_lib has to use halo2 library 

### PR DESCRIPTION
**Pull request by iquerejeta**

### CODE REVIEW

File: halo2_backend/src/helpers.rs [hunk 1]
```diff
@@ -37,9 +37,9 @@ pub trait SerdeCurveAffine: CurveAffine + SerdeObject {
     /// Reads an element from the buffer and parses it according to the `format`:
     /// - `Processed`: Reads a compressed curve element and decompress it
     /// - `RawBytes`: Reads an uncompressed curve element with coordinates in Montgomery form.
-    /// Checks that field elements are less than modulus, and then checks that the point is on the curve.
+    ///   Checks that field elements are less than modulus, and then checks that the point is on the curve.
```
Comment (miguel-ambrona) [inline halo2_backend/src/helpers.rs:40]: Is this intended?

**Review by miguel-ambrona — APPROVED** [review]
LGMT!

How about the zcash multiopen argument?

File: halo2_backend/src/helpers.rs (hunk 1 above)
Comment (iquerejeta) [reply halo2_backend/src/helpers.rs:40]: Yes. That's to align with the bullet points.

**Comment by iquerejeta** [comment]
The multi open argument from ZCash will be handled in a branch used for self-recursion. We still need to debug that.

**Pull request merged**